lib/arel/visitors/oracle12.rb

```diff
@@ -0,0 +1,53 @@
+module Arel
+  module Visitors
+    class Oracle12 < Arel::Visitors::ToSql
+      private
+
+      def visit_Arel_Nodes_SelectStatement o, collector
+        # Oracle does not allow LIMIT clause with select for update
+        if o.limit && o.lock
+          o = o.dup
+          o.limit = []
```

Comment:
This seems like a bug. If the db doesn't support `SELECT … FOR UPDATE` with a limit, we should raise an error (or use an alternative query?) rather than silently dropping the limit.